src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -132,6 +132,28 @@ public function findAvailableEpisodesByCategoryAncestry(
         return $this->resolveParents($result);
     }
 
+    public function findByCoreEntityMembership(int $entityId, ?string $groupType, ?int $limit, int $offset)
+    {
+        if ($groupType === null) {
+            $groupType = 'Group';
+        }
+
+        $qb = $this->getEntityManager()->createQueryBuilder()
+            ->select(['groupEntity', 'image'])
+            ->from('ProgrammesPagesService:' . $groupType, 'groupEntity')
+            ->join('ProgrammesPagesService:Membership', 'membership', Query\Expr\Join::WITH, 'membership.group = groupEntity')
+            ->leftJoin('groupEntity.image', 'image')
```

Comment:
Thinking about it again, please join to master brand, network and network image here, the domain object construction expects these things to be present and we don't want to be chasing obscure bugs for the sake of extra microseconds on the query,